gui/gui/src/ts/components/navbar/Navbar.tsx

```diff
@@ -1,12 +1,45 @@
 import * as React from "react";
 
-import {Typography, AppBar, Toolbar} from "@material-ui/core";
+import * as qs from "qs";
+
+import {
+  Typography,
+  AppBar,
+  Toolbar,
+  makeStyles,
+  createStyles,
+} from "@material-ui/core";
+
+import {NodeLabelSearch} from "components/search/NodeLabelSearch";
+import {useHistory} from "react-router-dom";
+import {Hrefs} from "Hrefs";
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    brand: {
+      marginRight: theme.spacing(2),
+    },
+  })
+);
 
 export const Navbar: React.FunctionComponent<{}> = () => {
+  const classes = useStyles();
+
+  const history = useHistory();
+
+  const onSearchSubmit = (text: string) => {
+    history.push(
+      Hrefs.nodeSearch + qs.stringify({text}, {addQueryPrefix: true})
+    );
+  };
+
   return (
     <AppBar position="static">
       <Toolbar>
-        <Typography variant="h6">Mowgli</Typography>
+        <Typography variant="h6" className={classes.brand}>
+          MOWGLI
+        </Typography>
+        <NodeLabelSearch onSubmit={onSearchSubmit} />
```

Comment:
This is fine for now. Later we'll support searching on multiple fields with Lucene queries. NodeSearchBox would have been better. Then we can potentially reuse it like on the conceptnet.io front page.

Reply:
Ah okay, that makes more sense. Ill just change it now